pkg/model/contextresource.go

```diff
@@ -0,0 +1,82 @@
+// Copyright 2022 The Okteto Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package model
+
+import (
+	"os"
+
+	"github.com/okteto/okteto/pkg/discovery"
+	oktetoErrors "github.com/okteto/okteto/pkg/errors"
+	"github.com/okteto/okteto/pkg/filesystem"
+	yaml "gopkg.in/yaml.v3"
+)
+
+// ContextResource provides the context and namespace to operate within a manifest
+type ContextResource struct {
+	Context   string
```

Comment:
The setters for these have some logic that helps provide a guard. Given that they should probably not be exported. Can you check if anyone is using this. If they are setting it without the setter then it may be we have a bug and if they are getting it then we need to add a getter.

Reply:
For marshaling within this package
``` 
// For private/package consumption due to yaml marshalling requirements.
type contextResource struct {
    Context string
    Namespace string
}

// For public consumption due to the requirement of a logical setter.
type ContextResource struct {
    context string
    namespace string
}
```

Reply:
That is something that needs to be addressed but I think this PR is big enough to add more changes. Changing that requires changing all `cmd/context` and their references so the number of touched files can go up to 30 files more. Could we do it as part of the `removing model` refactor?